routes/api/contacts.js

```diff
@@ -1,25 +1,101 @@
-const express = require('express')
+const express = require("express");
+const router = express.Router();
+const {
+  listContacts,
+  getContactById,
+  removeContact,
+  addContact,
+  updateContact,
+} = require("../../models/contacts");
+const Joi = require("@hapi/joi");
 
-const router = express.Router()
+const schemaPOST = Joi.object({
+  name: Joi.string().min(2).max(30).required(),
+  email: Joi.string().email().required(),
+  phone: Joi.string()
+    .pattern(/^[0-9()\s+-]+$/)
+    .min(8)
+    .max(30)
+    .required(),
+});
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const schemaPUT = Joi.object({
+  name: Joi.string().min(2).max(30).optional(),
+  email: Joi.string().email().optional(),
+  phone: Joi.string()
+    .pattern(/^[0-9()\s+-]+$/)
+    .min(8)
+    .max(30)
+    .optional(),
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  try {
+    const contacts = await listContacts();
+    res.status(200).json(contacts);
+  } catch (error) {
+    next(error);
+  }
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  const { contactId } = req.params;
+  try {
+    const contact = await getContactById(contactId);
+    res.status(200).json(contact);
+  } catch (error) {
+    res.status(404).json({ message: "Contact not found" });
+  }
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  try {
+    const { error, value } = schemaPOST.validate(req.body, {
+      stripUnknown: true,
+    });
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
+    const { name, email, phone } = value;
+    const newContact = await addContact({ name, email, phone });
+    res.status(201).json(newContact);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+});
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.delete("/:contactId", async (req, res, next) => {
+  const { contactId } = req.params;
+  try {
+    await removeContact(contactId);
+    res.status(200).json({ message: "contact deleted" });
+  } catch (error) {
+    res.status(404).json({ message: "Not found" });
```

Comment:
Podobnie tutaj, skąd wiesz że not found? A jeśli inny błąd to wprowadzisz siebie w błędne rozumienie. Jeśli już to next użyj.